src/activities/discussions/DiscussionTopicEntity.js

```diff
@@ -21,16 +21,58 @@ export class DiscussionTopicEntity extends Entity {
 	}
 
 	/**
-	 * Updates the discussion topic to have the given name
-	 * @param {string} name Name to set on the discussion topic
+	 * @summary Checks if topic entity has changed, primarily used for dirty check
+	 * @param {object} topic the topic that's being modified
 	 */
-	async setName(name) {
-		const action = this.canEditName() && this._entity.getActionByName(Actions.discussions.topic.updateName);
-		if (!action) {
-			return;
+	equals(topic) {
+		const diffs = [
+			[topic.name, this.name()],
+		];
+
+		for (const [current, initial] of diffs) {
+			if (current !== initial) {
+				return false;
+			}
 		}
 
+		return true;
+	}
+
+	/**
+	 * @summary Fires all the formatted siren actions collectively
```

Comment:
Love this.

Reply:
Thanks 😁 I renamed and tweaked a few things so it looks different from QuizEntity. I was worried that would be confusing!